### ISSUE TITLE
Does this work? :(

### ISSUE THREAD

**Issue by infinityplusb** [esthermations/DlangVulkanTriangle#1]
Hi, I've been looking for something that uses Vulkan with D and came across this project. Looks promising and the screenshot looks beautiful, but I can't get it to work.
Asides of manually compiling the shaders and bringing in my own cube.obj file, I have changed anything else.

dlang version: 2.1
vulkan: 1.3.211

**Comment by esthermations** [comment]
Hey thanks for your interest. It might be in a borked state right now, I'm not sure how I left it last time I worked on it.

When I get home tonight (~6h time) I'll see what's up with it. There will be commits in the history where it's working, but maybe I can tweak a thing or two and fix it up. Either way I'll let you know how to get it going. Thanks!

**Comment by esthermations** [comment]
So I've been plugging away at this over the weekend and it's not working on my laptop yet for... varying reasons.

I'll keep at it when I can since it's fun, but if you wanted a good starting point, I know this commit was working on my desktop PC at one point -- it's what I got the screenshot from: https://github.com/esthermations/DlangVulkanTriangle/commit/fd4404c73219a3d6350d62d1cb501b47033493be

**Comment by infinityplusb** [comment]
Ah yes, I had a poke around too.
I think I even tried to revert to that commit (or the next one which was the same with a readme), but maybe my dub and (more likely) Vulkan versions are different, as it was also giving me errors. 
I'll try revert the Vulkan version and see if that fixes things.

Thanks for your help. :)

**Comment by infinityplusb** [comment]
Is there something you referenced/followed etc to build out the renderer general game engine logic that I could look at to try and figure out wtf is going on ? :| 
I might be able to poke around and add some logic when we get it back up and running.

**Comment by esthermations** [comment]
> Is there something you referenced/followed etc to build out the renderer general game engine logic that I could look at to try and figure out wtf is going on ? :| 
> 
> I might be able to poke around and add some logic when we get it back up and running.

Yes, I think I was following this tutorial: https://vulkan-tutorial.com/

IIRC I took an opinionated approach to it and tried to straight-line the code where possible (hence having one huge Renderer class) rather than pop everything into separate objects, so my code will be a bit different. But the start-to-finish procedure should be essentially the same.

More recently I've been following the Intel API Without Secrets tutorial, which is also worth a look: https://www.intel.com/content/www/us/en/developer/articles/training/api-without-secrets-introduction-to-vulkan-preface.html
